### ISSUE TITLE
Perform benchmarks

### ISSUE THREAD

**Issue by hecrj** [hecrj/iced#7]
As mentioned in #4, in order to avoid performance regressions, it would be great to run some benchmarks on the different parts of the API.

The most expensive operation is clearly `UserInterface::build`. This will be the main focus for many of the optimizations we will implement in the future. We should benchmark this first.

`UserInterface::update` and `UserInterface::draw` will also get slower the more widgets and events a user interface has to handle. It would be great to have some numbers!

**Comment by TGotwig** [comment]
I can try to have a look 🕵🔎

**Comment by hecrj** [comment]
@TGotwig Awesome! I really appreciate the help. Let me know if you need anything.

**Comment by TGotwig** [comment]
CPU is constantly on `3.x` on MacOS, although nothing is happening.
<img width="815" alt="Aktivitätsanzeige (Meine Prozesse) 2019-09-14 14-47-26" src="https://user-images.githubusercontent.com/30773779/64908309-a16fb280-d6fe-11e9-860a-0c5cce31e80e.png">

**Comment by lachlansneff** [comment]
Looks to me like the event loop it's using is continuously polling.

**Comment by hecrj** [comment]
Yes, the example is drawing the UI every frame. This is expected.

Benchmarking the `ggez` example is not what this issue is about! `iced` is renderer-agnostic. We should benchmark `iced`.
